src/java/org/jivesoftware/openfire/spi/RoutingTableImpl.java

```diff
@@ -477,7 +478,11 @@ private boolean routeToRemoteDomain(JID jid, Packet packet,
 		        }
 		    }
 		}
-		else {
+		else if (!RemoteServerManager.canAccess(jid.getDomain())) { // Check if the remote domain is in the blacklist
+            Log.info( "Unable to authenticate: Remote domain is not accessible according to our configuration (typical causes: server federation is disabled, or domain is blacklisted)." );

```

Comment:
Yes... I just copied the existing error message, but I think that must have the domain in the logger. I'll fix this.